spec/producer_spec.rb

```diff
@@ -140,5 +142,53 @@
       # The producer was not able to write the message, but it's still buffered.
       expect(producer.buffer_size).to eq 0
     end
+
+    context "when a partitioner is not specified" do
```

Comment:
I'm not overly happy with how these specs turned out. Suggestions are very welcome here.